src/echoclient/EchoClient.php

```diff
@@ -62,51 +66,62 @@ function __construct()
     }
 
     /**
-     * Add an event to echo
+     * Adds a single event to echo
      *
      * @param string $class
      * @param string $source
      * @param array $props
      * @param string|null $userId
+     * @param string|null $timestamp
      * @return bool True if successful, else false
      */
-    public function createEvent($class, $source, array $props = array(), $userId = null)
+    public function createEvent($class, $source, array $props = array(), $userId = null, $timestamp = null)
     {
-        $class = ECHO_CLASS_PREFIX . $class;
-        $baseUrl = $this->getBaseUrl();
-
-        if (!$baseUrl) {
-            // fail silently when creating events, should not stop user interaction as echo events are collected on a best-endeavours basis
-            $this->getLogger()->warning('Echo server is not defined (missing ECHO_HOST define), not sending event - ' . $class,
-                $props);
+        try {
+            $event = new \echoclient\EchoEvent($class, $source, $props, $userId, $timestamp);
+            return $this->sendBatchEvents([$event]);
+        } catch (\Exception $e) {
             return false;
         }
+    }
 
-        $eventUrl = $baseUrl . '/events';
-        $eventJson = $this->getEventJson($class, $source, $props, $userId);
+    /**
+     * Adds multiple events to echo
+     *
+     * @param EchoEvent[] $events An array of EchoEvent objects
+     * @return bool True if successful
+     * @throws PayloadTooLargeException If the size, in bytes, of batched events exceeds configured limit
```

Comment:
can we explicitly state the limit?  I know it's a constant, but if it's in the docs, it helps the dev, as per Mark's concerns about it be suboptimal.

Reply:
(or at least reference the variable, so someone can see what the limit is...?)